src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -245,129 +245,185 @@ void StateControllerImpl::SetRegularState(ApplicationSharedPtr app,
 }
 
 namespace {
-bool HasStreamingPermissions(mobile_apis::HMILevel::eType val) {
+bool IsStreamableHMIType(mobile_apis::HMILevel::eType val) {
   using namespace helpers;
   return Compare<mobile_apis::HMILevel::eType, EQ, ONE>(
       val, mobile_apis::HMILevel::HMI_FULL, mobile_apis::HMILevel::HMI_LIMITED);
 }
+
+bool IsSameAudioAppType(const Application& app1, const Application& app2) {
+  // compatibility constants
+  const bool both_navi = app1.is_navi() && app2.is_navi();
+  const bool both_vc = app1.is_voice_communication_supported() &&
+                       app2.is_voice_communication_supported();
+  const bool both_media = (app1.is_media_application() && !app1.is_navi() &&
```

Comment:
@EKuliiev May be use utils::Compare instead?

Reply:
@AByzhynar see: 39d2bceb5b4040aa09e173314c4a9a3c078bdfa0
Agree with all review issues except this. Determining of isMediaAppType moved in separate func.